### DISCUSSION TITLE
Creating a common many-to-many table for models based on differentiator column

### DISCUSSION THREAD

**Discussion by zhulien-ivanov** [prisma/prisma#19142] [Q&A] [answered]
I have a `Tag`, `User` and `Role` entities. I want `User` and `Role` to have relation to tags. Instead of creating 2 (or n) many-to-many tables supporting that, I want to create a single table that hosts all tag relations and differentiates them by entity type. Here is an example:

| id          | entity-type    | entity-id | tag-id |
| -------- | -------------- | ---------  | -------|
| 1           | user               | 240         | 14      |
| 2           | role               | 32            | 9        |
| 3           | role               | 32            | 14      |

Where `entity-type` is the differentiator and `entity-id` is the id of the specific entity. The question is, how (if possible) can I represent this in a prisma model? So, any `User` should have relation to all tags from the many-to-many table that are of `entity-type` = `"user"` and have the appropriate id.

**Comment by ajmnz** [comment, marked as answer]
This sounds like a polymorphic association, which is [not yet supported natively by Prisma](https://github.com/prisma/prisma/issues/1644). There are some elegant solutions in the linked issue, but nothing you can do _within_ Prisma.

**Reply by ludralph** [reply]
Hey @zhulien-ivanov ,

do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Reply by zhulien-ivanov** [reply]
Thank you, @ajmnz. Went over the thread and I'm surprised this is an outstanding issue for more than 2 years now without a solution, considering it is an everyday design & development case. Basically, the least bad workaround (as we can't really talk about elegant solutions) is to join your polymorphic table manually with a second request.
